app/Http/Controllers/OAuthController.php

```diff
@@ -2,20 +2,78 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\User;
-use CDash\Middleware\OAuth2\OAuth2Interface;
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\RedirectResponse as symfonyResponse;
 use RuntimeException;
 
+use App\Models\User;
+use App\Http\Controllers\Auth\RegisterController;
+use CDash\Middleware\OAuth2\OAuth2Interface;
+
 /**
  * Class OAuthController
  * @package App\Http\Controllers
  */
 final class OAuthController extends AbstractController
 {
+    /**
+     * @param Request $request
+     * @param string $service
+     * @return symfonyResponse
+     */
+
+    public function socialite(Request $request, string $service)
+    {
+        return Socialite::driver($service)->redirect();
+    }
+
+    /**
+     * @param string $service
+     * @param User $user_table
+     * @return RedirectResponse
+    */
```

Comment:
PHPDoc comments aren't needed unless you are using them to provide extra information to PHPStan.  These comments duplicate the native type annotations you already have.